win32ss/user/ntuser/menu.c

```diff
@@ -2992,10 +2992,8 @@ static BOOL FASTCALL MENU_ShowPopup(PWND pwndOwner, PMENU menu, UINT id, UINT fl
     /* We are off the right side of the screen */
     if (x + width > monitor->rcMonitor.right)
     {
-        if ((x - width) < monitor->rcMonitor.left || x >= monitor->rcMonitor.right)
-            x = monitor->rcMonitor.right - width;
-        else
-            x -= width;
+        /* Position menu at right edge of screen */
+        x = monitor->rcMonitor.right - width;
```

Comment:
Serges question about right column is acceptable and maybe Doug can answer it. But regarding the "bottom"-discussion I want to intervene: **Serge, Let's not touch any code within this PR, that is not proven by a testcase to be broken**. Mark Jansen spent several years into getting the menu-positioning-code into shape.

Reply:
What "right" column are you refering to @SergeGautherie

Reply:
@SergeGautherie can you do a suggestion what you want to see changed? Do you want any specific comment to be changed?

Reply:
@SergeGautherie, I think your 'right column' comment was actually about the 'rightmost pixel' and after thinking about this, I believe that you are correct. My formula needed to be incremented by one to get to the exact rightmost pixel. For example if our width was 3 and our 'monitor->rcMonitor.right' was 768, then x = 768 - 3 = 765. So our 3 pixels would be in x-pixel positions of 765, 766, and 767, meaning that we did not reach the last one of 768. I have made a correction for this now. Thanks.

Reply:
As on the previous PR, I haven't checked (yet) how WS2003 behaves. (It would be nice to have an automated test for this feature.)
My comments are only to make PR(s) self-consistent at least.

Reply:
After more testing, I reverted my previous commit and added some explanations. Thanks.